source/server/http/admin.cc

```diff
@@ -128,6 +148,9 @@ Http::Code AdminImpl::handlerClusters(const std::string&, Buffer::Instance& resp
           fmt::format("{}::{}::canary::{}\n", cluster.second->name(), host->url(), host->canary()));
       response.add("\n");
     }
+    if (cluster.second->hosts().empty()) {
```

Comment:
I would just remove this new code as well as the newline in the hosts case, as well as the === === thing above. The reality is that this is always used with grep so it's all pointless.